osgi.specs/docbook/160/service.jakartaws.xml

```diff
@@ -0,0 +1,480 @@
+
+          <thead>
+            <row>
+              <entry>Service Property Name</entry>
+
+              <entry>Type</entry>
+
+              <entry>Description</entry>
+            </row>
+          </thead>
+
+          <tbody>
+            <row>
+              <entry><code>service.changecount</code></entry>
+              <entry><code>Long</code></entry>
+              <entry><para>Whenever the DTOs available from the WebserviceServiceRuntime service change, the value
+              of this property will increase.</para> <para>This allows
+              interested parties to be notified of changes to the DTOs by
+              observing Service Events of type <code>MODIFIED</code> for the
+              <xref linkend="org.osgi.service.webservice.runtime.WebserviceServiceRuntime"
+              xrefstyle="hyperlink"/> service. See <code>org.osgi.framework.Constants.SERVICE_CHANGECOUNT</code> in <xref linkend="intro.core.release" xrefstyle="template:%t"/>.</para></entry>
+            </row>
+          </tbody>
+        </tgroup>
+      </table>
+        </section>
+        <section>
+            <title>Whiteboard Error Handling</title>
+        
+            <para>
+            There are a number of error cases where the Jakarta Webservice
+            whiteboard may be unable to correctly register an endpoint or use a given handler.
+            All of these cases must result in a failure DTO being created with the
+            appropriate error code.
+            </para>
+            <section>
+                <title>General Error Codes</title>
+                <itemizedlist>
+                  <!-- generic error codes -->    
+                  <listitem>
+                    <para>
+                    <emphasis>Unknown Failure</emphasis> - Outside of the predefined error conditions there might still errors occurring during the construction of the objects,
+                    the implementation can use the error code <code>0</code> to indicate that.
+                    The implementation can use own error codes to but to prevent clash with codes specified in later specification versions, such an error code must
+                    be equal or larger than <code>1000</code>.
+                    In all cases an additional error message might be used to further describe the nature of the failure and give users of the DTO a hint what's going on.
+                    </para>
+                  </listitem>
+                  <listitem>
+                    <para><emphasis>Failure to obtain a service instance</emphasis> - In
+                    the case where a published service is unable to be obtained by the
+                    Jakarta Webservice whiteboard then the service is deny
+                    listed by the container. A failure DTO is made available from the
+                    WebserviceServiceRuntime representing the deny listed service
+                    object. The error code in this case is <code>1</code>.</para>
```

Comment:
There should be constants for the defined error codes. See https://github.com/osgi/osgi/blob/main/org.osgi.service.servlet/src/org/osgi/service/servlet/runtime/dto/DTOConstants.java as an example

Reply:
See https://github.com/osgi/osgi/pull/659/files#diff-3156a92db7e047bef4da747c15e1ffb4a7a9f0754cd1eea27b2d815601e84362R36

Reply:
It's good that they're there already. Please link them in the spec text so that they're easier to find 😃

Reply:
Done